test/dest.js

```diff
@@ -763,53 +763,6 @@ describe('dest stream', function() {
     stream.end();
   });
 
-  it('should write file atime and mtime using the vinyl stat', function(done) {
```

Comment:
```sh
$ touch in out

$ ls -lu in out #atime
-rw-r--r--  1 erik  staff  0 Jan 15 22:25 in
-rw-r--r--  1 erik  staff  0 Jan 15 22:25 out

$ ls -lc in out #mtime
-rw-r--r--  1 erik  staff  0 Jan 15 22:25 in
-rw-r--r--  1 erik  staff  0 Jan 15 22:25 out

$ #wait a minute
$ cat in > out

$ ls -lu in out #atime
-rw-r--r--  1 erik  staff  0 Jan 15 22:26 in
-rw-r--r--  1 erik  staff  0 Jan 15 22:25 out
ls -lc in out #mtime
-rw-r--r--  1 erik  staff  0 Jan 15 22:25 in
-rw-r--r--  1 erik  staff  0 Jan 15 22:26 out
```

Currently, the `dest` code updates the `atime` on the vinyl file because it gets its `contents`. While the `src` code updates the `mtime` on the vinyl file because it is setting its `contents`, and so will always appear to be modified.

That seems wrong externally (the `out` file will have the wrong `atime`) and internally (plugins will get a later `mtime` than the sourced file).

Furthermore, questions of what we're allowed to do privilege-wise are different if we are creating or overwriting the destination, which is a distinction which is implicit, at best, as things now stand.